src/terminal/input/terminalInput.cpp

```diff
@@ -466,6 +468,44 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent) const
     return fKeyHandled;
 }
 
+bool TerminalInput::HandleChar(const char16_t ch)
+{
+    if (ch == UNICODE_BACKSPACE || ch == UNICODE_DEL)
+    {
+        return false;
+    }
+    else if (Utf16Parser::IsLeadingSurrogate(ch))
+    {
+        if (_leadingSurrogate.has_value())
+        {
+            // we already were storing a leading surrogate but we got another one. Go ahead and send the
+            // saved surrogate piece and save the new one
+            wchar_t buffer[32];

```

Comment:
How did you select 32 wchar_ts as the maximum buffer size? It seems overly large for a surrogate between U+D800 and U+DFFF (55,296 to 57,343).

Reply:
I swear I got this from the surrogate code I moved over from the original terminal control [here](https://github.com/microsoft/terminal/pull/2004/files/dd48401410319a4941331232cbae3c447c424828#diff-37b8cdfbaef0c1f807e324f0e6efb4edL586-L617), but I must be mistaken. I'll shorten this up.

Reply:
oh, I found it. The old code used `to_hstring` and internally that uses a buffer that is 32 `wchar_t`s large

Reply:
Ah, ok.